hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsStatistics.java

```diff
@@ -0,0 +1,280 @@
+  @Test
+  public void testCreateStatistics() throws IOException {
+    describe("Testing counter values got by creating directories and files in"
+        + " Abfs");
+
+    AzureBlobFileSystem fs = getFileSystem();
+    Path createFilePath = path(getMethodName());
+    Path createDirectoryPath = path(getMethodName() + "Dir");
+
+    Map<String, Long> metricMap = fs.getInstrumentation().toMap();
+
+    /*
+     Test for initial values of create statistics ; getFileStatus is called
+     1 time after Abfs initialisation.
+     */
+    assertEquals("Mismatch in op_create", 0,
+        (long) metricMap.get("op_create"));
+    assertEquals("Mismatch in op_create_non_recursive", 0,
+        (long) metricMap.get("op_create_non_recursive"));
+    assertEquals("Mismatch in files_created", 0,
+        (long) metricMap.get("files_created"));
+    assertEquals("Mismatch in directories_created", 0,
+        (long) metricMap.get("directories_created"));
+    assertEquals("Mismatch in op_mkdirs", 0,
+        (long) metricMap.get("op_mkdirs"));
+    assertEquals("Mismatch in op_get_file_status", 1,
+        (long) metricMap.get("op_get_file_status"));
+
+    fs.mkdirs(createDirectoryPath);
+    fs.createNonRecursive(createFilePath, FsPermission
+        .getDefault(), false, 1024, (short) 1, 1024, null);
+
+    metricMap = fs.getInstrumentation().toMap();
+      /*
+       Test of statistic values after creating a directory and a file ;
+       getFileStatus is called 1 time after creating file and 1 time at
+       time of initialising.
+       */
+    assertEquals("Mismatch in op_create", 1,
+        (long) metricMap.get("op_create"));
+    assertEquals("Mismatch in op_create_non_recursive", 1,
+        (long) metricMap.get("op_create_non_recursive"));
+    assertEquals("Mismatch in files_created", 1,
+        (long) metricMap.get("files_created"));
+    assertEquals("Mismatch in directories_created", 1,
+        (long) metricMap.get("directories_created"));
+    assertEquals("Mismatch in op_mkdirs", 1,
+        (long) metricMap.get("op_mkdirs"));
+    assertEquals("Mismatch in op_get_file_status", 2,
+        (long) metricMap.get("op_get_file_status"));
+
+    //re-initialising Abfs to reset statistic values.
+    fs.initialize(fs.getUri(), fs.getConf());
+
+      /*Creating 10 directories and files; Directories and files can't
+       be created with same name, hence <Name> + i to give unique names.
+       */
+    for (int i = 0; i < NUMBER_OF_OPS; i++) {
+      fs.mkdirs(path(getMethodName() + "Dir" + i));
+      fs.createNonRecursive(path(getMethodName() + i),
+          FsPermission.getDefault(), false, 1024, (short) 1,
+          1024, null);
+    }
+
+    metricMap = fs.getInstrumentation().toMap();
+      /*
+       Test of statistics values after creating 10 directories and files;
+       getFileStatus is called 1 time at initialise() plus number of
+       times file is created.
+       */
+    assertEquals("Mismatch in op_create", NUMBER_OF_OPS,
```

Comment:
Should we create a new method assertValue(String operation, long expected, long actual) and refactor?

Reply:
I think we had that earlier but decided to use assertEquals instead because assertValue was essentially a method to call assertEquals and we could directly call assertEquals too?